test/services/SubmissionServiceSpec.scala

```diff
@@ -41,156 +43,286 @@ class SubmissionServiceSpec
           mockCreateBusinessVerificationJourney(testJourneyId, testSautr)(Future.successful(Left(NotEnoughEvidence)))
           mockStoreBusinessVerificationStatus(testJourneyId, BusinessVerificationUnchallenged)(Future.successful(SuccessfullyStored))
           mockStoreRegistrationResponse(testJourneyId, RegistrationNotCalled)(Future.successful(SuccessfullyStored))
           mockCreateTrn(testJourneyId)(Future.successful(testTrn))
 
-          val result = await(TestService.submit(testJourneyId))
+          val result = await(TestService.submit(testJourneyId, testSoleTraderJourneyConfig))
 
           result mustBe JourneyCompleted(testContinueUrl)
 
         }
         "no sautr is provided" in {
           enable(EnableNoNinoJourney)
-          mockGetJourneyConfig(testJourneyId)(Future.successful(testJourneyConfig(true)))
           mockRetrieveIndividualDetails(testJourneyId)(Future.successful(Some(testIndividualDetailsNoNinoNoSautr)))
           mockMatchSoleTraderDetailsNoNino(testJourneyId, testIndividualDetailsNoNinoNoSautr)(Future.successful(Right(true)))
           mockStoreBusinessVerificationStatus(testJourneyId, BusinessVerificationUnchallenged)(Future.successful(SuccessfullyStored))
           mockStoreRegistrationResponse(testJourneyId, RegistrationNotCalled)(Future.successful(SuccessfullyStored))
           mockCreateTrn(testJourneyId)(Future.successful(testTrn))
 
-          val result = await(TestService.submit(testJourneyId))
+          val result = await(TestService.submit(testJourneyId, testSoleTraderJourneyConfig))
 
           result mustBe JourneyCompleted(testContinueUrl)
         }
       }
       "return SoleTraderDetailsMismatch" when {
         "the details received from ES20 do not match" in {
-          mockGetJourneyConfig(testJourneyId)(Future.successful(testJourneyConfig(true)))
           mockRetrieveIndividualDetails(testJourneyId)(Future.successful(Some(testIndividualDetailsNoNino)))
           mockMatchSoleTraderDetailsNoNino(testJourneyId, testIndividualDetailsNoNino)(Future.successful(Left(DetailsMismatch)))
           mockStoreIdentifiersMatch(testJourneyId, identifiersMatch = false)(Future.successful(SuccessfullyStored))
           mockStoreBusinessVerificationStatus(testJourneyId, BusinessVerificationUnchallenged)(Future.successful(SuccessfullyStored))
           mockStoreRegistrationResponse(testJourneyId, RegistrationNotCalled)(Future.successful(SuccessfullyStored))
 
-          val result = await(TestService.submit(testJourneyId))
+          val result = await(TestService.submit(testJourneyId, testSoleTraderJourneyConfig))
 
           result mustBe SoleTraderDetailsMismatch(DetailsMismatch)
         }
       }
     }
-    "the user has no nino on the Individual Journey" should {
+
+    "for individual journey: the user has no nino and matching" should {
       "return JourneyCompleted" when {
         "no nino is provided" in {
```

Comment:
I don't think we need this line anymore "no nino is provided". The test is described well enough in this lines above it

Reply:
yes agreed